pkg/config/config_template.yaml

```
@@ -413,6 +413,126 @@ api_key:
+  ## The number of concurrent tasks used to discover SNMP devices. Increasing this value
+  ## discovers devices faster but at the cost of increased resource consumption.
+  #
+  # workers: 2
+
+  ## @param discovery_interval - integer - optional - default: 3600
+  ## How often to discover new SNMP devices, in seconds. Decreasing this value
+  ## discovers devices faster but at the cost of increased resource consumption.
+  #
+  # discovery_interval: 3600
+
+  ## @param allowed_failures - integer - optional - default: 3
+  ## The number of failed requests to a given SNMP device before removing it from the list of monitored
+  ## devices.
+  ## If a device shuts down, the Agent stops monitoring it after `discovery_interval * allowed_failures` seconds.
+  #
+  # allowed_failures: 3
+
+  ## @param configs - list - required
+  ## The actual configurations used to discover SNMP devices in various subnets.
+  #
+  # configs: 
+    ## @param network - string - required
+    ## The subnet in CIDR format to scan for SNMP devices.
+    ## All unignored IP addresses in the CIDR range are scanned.
+    ## For optimal discovery time, be sure to use the smallest network mask
+    ## possible as is appropriate for your network topology.
+    ## Ex: 10.0.0.0/24
+    #
+    # - network: <NETWORK>
+
+    ## @param ignored_ip_addresses - list of strings - optional
+    ## A list of IP addresses to ignore when scanning the network.
+    #
+    # ignored_ip_addresses:
+    #   - <IP_ADDRESS_1>
+    #   - <IP_ADDRESS_2>
+
+    ## @param port - integer - optional - default: 161
+    ## The UDP port to use when connecting to SNMP devices.
+    #
+    # port: 161
+
+    ## @param version - integer - optional - default: <BEST_GUESS>
+    ## Set the version of the SNMP protocol. Value can be `1`, `2` or `3`.
+    ## If unset, the Agent tries to guess the correct version based on other configuration
+    ## parameters. i.e if `user` is set, the Agent will use SNMP v3.
+    #
+    # version: <VERSION>
+
+    ## @param timeout - integer - optional - default: 5
+    ## The number of seconds before timing out.
+    #
+    # timeout: 5
+
+    ## @param retries - integer - optional - default: 3
+    ## The number of retries before failure.
+    #
+    # retries: 3
+
+    ## @param community - string - optional
+    ## Required for SNMP v1 & v2.
+    ## Ex: 'public'
+    #
+    # community: <COMMUNITY_STRING>
+
+    ## @param user - string - optional
+    ## The username to connect to your SNMP devices.
+    ## SNMPv3 only.
+    #
+    # user: <USERNAME>
+
+    ## @param authentication_key - string - optional
+    ## The passphrase to use with your Authentication type.
+    ## SNMPv3 only.
+    #
+    # authentication_key: <AUTHENTICATION_TYPE_KEY>
+
+    ## @param authentication_protocol - string - optional
+    ## Authentication protocol to use when connecting to your SNMP devices.
+    ## Available options are: MD5, SHA, SHA224, SHA256, SHA384, SHA512.
```

Due to #5652 currently we only have `MD5` and `SHA`.

```suggestion
    ## Available options are: MD5, SHA.
```

---

arf thanks I didn't see that revert PR